### ISSUE TITLE
NPM install doesn't work.

### ISSUE THREAD

**Issue by stampeder** [bpatrik/pigallery2#112]
root@piphoto:~/pigallery2-master# npm install
(node:4235) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
npm ERR! Error: Method Not Allowed
npm ERR!     at errorResponse (/usr/share/npm/lib/cache/add-named.js:260:10)
npm ERR!     at /usr/share/npm/lib/cache/add-named.js:120:12
npm ERR!     at saved (/usr/share/npm/node_modules/npm-registry-client/lib/get.js:167:7)
npm ERR!     at FSReqWrap.oncomplete (fs.js:135:15)
npm ERR! If you need help, you may report this *entire* log,
npm ERR! including the npm and node versions, at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! System Linux 4.19.66-v7+
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! cwd /root/pigallery2-master
npm ERR! node -v v8.11.1
npm ERR! npm -v 1.4.21
npm ERR! code E405
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /root/pigallery2-master/npm-debug.log
npm ERR! not ok code 0

The above is the console output from the code I downloaded from this Git. I'm running Raspian Stretch fully updated and upgraded.

**Comment by bpatrik** [comment]
Hi,

By the look it seems that you are using node v8 that is officially not supported. 
v12 is recommended. Can you pls try with that? (package manager node installation installs an old node version)

Best,
Patrik

**Comment by stampeder** [comment]
Just to be clear the instructions for install state:


        1.1.1-a Install from release

cd  ~
wget https://github.com/bpatrik/pigallery2/releases/download/1.7.0/pigallery2.zip
unzip pigallery2.zip -d pigallery2
cd  pigallery2
npm install


        <https://github.com/bpatrik/pigallery2#111-b-install-from-source>1.1.1-b
        Install from source

cd  ~
wget https://github.com/bpatrik/pigallery2/archive/master.zip
unzip master.zip
cd  pigallery2-master# enter the unzipped directory
npm install
npm build

*Note*: if you run|npm run create-release|, it creates a clean, 
minified, production ready version from the app in the|release|folder, 
that is ready to deploy.

*Note2*: you can use|npm run create-release -- --languages=fr,ro|to 
restrict building to the listed languages (english is added by default)


        <https://github.com/bpatrik/pigallery2#112-run-pigallery2>1.1.2
        Run PiGallery2

npm start

Thus to install node v12 I would have to not use npm install or nmp 
build as it install v8? But inorder to start PiGallery2 I would still 
have to use npm start to get it going?

Thanks for the response and the awesome work you are doing on this. I am 
really stoked to get it going.
Glenn.

**Comment by bpatrik** [comment]
Hi,

First of all, I'm currently working on a major update to the app and there might be some bugs on the master branch. There were also several breaking changes compared to v1.7 (v1.8 should be released during the next week)
Until then I recommend using v1.7.0.

There error, you have posted doesn't really contain any information, but it seems that you are using node v8. I'm not sure what happens with node  v8. And also that is a bit outdated.
Here is a tutorial for installing node: https://thisdavej.com/beginners-guide-to-installing-node-js-on-a-raspberry-pi/
In short, you most likely need this: `curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -`

Once you have node v12, you can install the release:
```
cd ~
wget https://github.com/bpatrik/pigallery2/releases/download/1.7.0/pigallery2.zip
unzip pigallery2.zip -d pigallery2
cd pigallery2
npm install
```

Best,
Patrik

**Comment by bpatrik** [comment]
I'm officially supporting docker (and docker-compose) now. 
I recommend using that for installations.

**Comment by bpatrik** [comment]
closing due to inactivity
